packages/web/components/one-click-trading/one-click-trading-settings.tsx

```diff
@@ -105,13 +106,18 @@ export const OneClickTradingSettings = ({
   hasExistingSession,
   onEndSession,
   onClose,
+  resetTransaction1CTParams,
+  showCreationButton = true,
 }: OneClickTradingSettingsProps) => {
   const { t } = useTranslation();
   const [changes, setChanges] = useState<
     Array<"spendLimit" | "resetPeriod" | "sessionPeriod">
   >([]);
-  const [initialTransaction1CTParams, setInitialTransaction1CTParams] =
-    useState<OneClickTradingTransactionParams>();
+
+  const handleOnStartSession = () => {
+    onStartTrading();
+    setChanges([]);
+  };
```

Comment:
The component stay generic, this is an improvement. The component before already handle onStartTrading logic, what I did is make sure that after trigger the tx, clean the state changes. It was working where was being used before because the onGoBack was unmounting the component and dropping the state. However there could be scenarios where the component is not unmounted and we need to make sure that changes are cleaned.